packages/gatsby/src/redux/actions/public.ts

```diff
@@ -1112,21 +1143,25 @@ actions.replaceWebpackConfig = (config: Object, plugin?: ?Plugin = null) => {
  *   }
  * })
  */
-actions.setBabelOptions = (options: Object, plugin?: ?Plugin = null) => {
+actions.setBabelOptions = (
+  options: Record<string, any>,
+  plugin: IGatsbyPlugin | null = null
+): ISetBabelOptionsAction => {
   // Validate
-  let name = `The plugin "${plugin.name}"`
-  if (plugin.name === `default-site-plugin`) {
-    name = `Your site's "gatsby-node.js"`
-  }
-  if (!_.isObject(options)) {
+  const pluginPlaceholder = `Your site's "gatsby-node.js"`
+  const hasNoPluginName = !plugin || plugin.name === `default-site-plugin`
+  const name = `The plugin "${
+    hasNoPluginName ? pluginPlaceholder : plugin!.name
+  }"`
+  if (typeof options !== `object`) {
```

Comment:
this technically is different. `options = []` would pass this.

Reply:
Lodash's own documentation of `isObject` says:

```
Checks if value is the language type of Object. (e.g. arrays, functions, objects, regexes, new Number(0), and new String(''))
```

So arrays would still pass this, should I add code that checks for that?